login-workflow/src/new-architecture/screens/CreateAccountScreen/CreateAccountScreen.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useCallback, useState } from 'react';
+import { CreateAccountScreenProps } from './types';
+import { CreateAccountScreenBase } from './CreateAccountScreenBase';
+import { useLanguageLocale } from '../../hooks';
+import { useRegistrationContext } from '../../contexts/RegistrationContext/context';
+import { useRegistrationWorkflowContext } from '../../contexts';
+
+/**
+ * Component that renders a screen for the user to enter their email address to start the
+ * account creation process.
+ *
+ * @param emailLabel label for the textfield
+ * @param initialValue used to pre-populate the email input field
+ * @param emailValidator used to test the input for valid formatting
+ * @param emailTextFieldProps props to pass to the email text field
+ *
+ * @category Component
+ */
+const EMAIL_REGEX = /^[A-Z0-9._%+'-]+@[A-Z0-9.-]+\.[A-Z]{2,}$/i;
+
+export const CreateAccountScreen: React.FC<CreateAccountScreenProps> = (props) => {
+    const { t } = useLanguageLocale();
+    const { actions } = useRegistrationContext();
+    const regWorkflow = useRegistrationWorkflowContext();
+    const { nextScreen, previousScreen, screenData } = regWorkflow;
+    const [emailInputValue, setEmailInputValue] = useState(screenData.CreateAccount.emailAddress);
+    const [isLoading, setIsLoading] = useState(false);
+
+    const onNext = useCallback(async () => {
+        try {
+            setIsLoading(true);
+            await actions().requestRegistrationCode(emailInputValue);
+            nextScreen({
+                screenId: 'CreateAccount',
+                values: { emailAddress: emailInputValue },
+            });
+            setIsLoading(false);
+        } catch {
+            console.error('Error while updating create account!');
```

Comment:
Need to setIsLoading(false) for error block also or else screen will freeze with loader.
![Uploading Screenshot 2023-07-05 at 11.13.59 AM.png…]()